src/azure-cli/azure/cli/command_modules/network/_format.py

```diff
@@ -139,14 +139,14 @@ def transform_nsg_rule_table_output(result):
     item['Priority'] = result['priority']
     item['SourcePortRanges'] = result['sourcePortRange'] or ' '.join(result['sourcePortRanges'])
     item['SourceAddressPrefixes'] = result['sourceAddressPrefix'] or ' '.join(result['sourceAddressPrefixes'])
-    item['SourceASG'] = result['sourceApplicationSecurityGroups'] or 'None'
+    item['SourceASG'] = result.get('sourceApplicationSecurityGroups', 'None')
     item['Access'] = result['access']
     item['Protocol'] = result['protocol']
     item['Direction'] = result['direction']
     item['DestinationPortRanges'] = result['destinationPortRange'] or ' '.join(result['destinationPortRanges'])
```

Comment:
Do we need to fix them as well?

Reply:
Nice catch! https://github.com/Azure/azure-cli/pull/25388 will cover it.